bridge/src/main/java/com/livefront/bridge/wrapper/WrapperUtils.java

```diff
@@ -0,0 +1,39 @@
+package com.livefront.bridge.wrapper;
+
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.os.Parcel;
+
+import java.util.Set;
+
+/**
+ * Handles the wrapping and unwrapping of certain {@link android.os.Parcelable} objects that use
+ * native code to optimize their {@code Parcelable} implementations. When placed into a {@link
+ * Bundle} unwrapped, these objects will cause a crash if the {@code Bundle} is written to a
+ * {@link android.os.Parcel} that then calls {@link Parcel#marshall()}.
+ */
+public class WrapperUtils {
+
+    public static void unwrapOptimizedObjects(Bundle bundle) {
+        Set<String> keys = bundle.keySet();
+        for (String key : keys) {
+            if (bundle.get(key) instanceof BitmapWrapper) {
```

Comment:
Ah gotcha. Yes that is the idea. But I like your wrapper idea, although it would only go one way (the unwrapping way); you'd still have to check for each individual object type (like `Bitmap`) when going to wrap it. But I like the idea....I think its worth looking into if I end up starting to wrap more types.